packages/slate/src/interfaces/editor/unhangRange.ts

```diff
@@ -19,14 +18,19 @@ export type UnhangRangeOptions = EditorUnhangRangeOptions & {
  * - `unhang` is true,
  * - `at` (default: selection) is a range.
  */
-export const unhangRange = <V extends Value>(
+export const unhangRange = <
+  V extends Value,
+  R extends Range | Path | Point | Span | null | undefined,
+>(
   editor: TEditor<V>,
-  range?: Range | Path | Point | Span | null,
+  range: R,
   options: UnhangRangeOptions = {}
-) => {
+): R => {
   const { voids, unhang = true } = options;
 
   if (Range.isRange(range) && unhang) {
-    return Editor.unhangRange(editor as any, range, { voids }) as TRange;
+    return Editor.unhangRange(editor as any, range, { voids }) as R;
   }
+
+  return range;
```

Comment:
Small change in behaviour: `unhangRange` now returns the input for non-ranges, or when the input is not a range. Previously, it returned undefined.